st2api/st2api/controllers/resource.py

```diff
@@ -159,10 +159,19 @@ def _get_all(self, exclude_fields=None, sort=None, offset=0, limit=None, query_o
         if offset >= 2**31:
             raise ValueError('Offset "%s" specified is more than 32-bit int' % (offset))
 
-        if limit and int(limit) > self.max_limit:
-            # TODO: We should throw here, I don't like this.
-            msg = 'Limit "%s" specified, maximum value is "%s"' % (limit, self.max_limit)
-            raise ValueError(msg)
+        if limit:
```

Comment:
Do we want to support `limit=0` and if yes, what should we do in such scenario? Does limit=0 even make sense?

Reply:
Nope, but then default was always `limit=0`, now we are going `limit=-1` route. How should we block `limit=0` i.e ask user to use -1 or some positive value or throw some other message? And then with RBAC support we need to provide non-admin user a message if they use `limit=-1`

Reply:
Yeah, maybe just throw / return an error message on limit=0 in query params.